hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java

```diff
@@ -204,6 +217,157 @@ private void fetchWebService(String urlStr) {
+        }
+
+        MCPhoneHomeInfo mcPhoneHomeInfo = getManCenterPhoneHomeInfo(mcResponse);
+        String version = mcPhoneHomeInfo.getVersion();
+        Object license = mcPhoneHomeInfo.getLicense();
+        int responseCode = mcPhoneHomeInfo.getResponseCode();
+        if (responseCode == HttpURLConnection.HTTP_OK) {
+            if (license == null) {
+                checkClusterSizeAndSetLicense(clusterSize, parameterCreator);
+            } else {
+                parameterCreator.addParam("mclicense", license.toString());
+            }
+            parameterCreator.addParam("mcver", version);
+        } else {
+            parameterCreator.addParam("mclicense", "MC_CONN_ERR_" + responseCode);
+            parameterCreator.addParam("mcver", "MC_CONN_ERR_" + responseCode);
+        }
+    }
+
+    private void checkClusterSizeAndSetLicense(int clusterSize, PhoneHomeParameterCreator parameterCreator) {
+        if (clusterSize <= 2) {
+            parameterCreator.addParam("mclicense", "MC_LICENSE_NOT_REQUIRED");
+        } else {
+            parameterCreator.addParam("mclicense", "MC_LICENSE_REQUIRED_BUT_NOT_SET");
+        }
+    }
+
+    private MCPhoneHomeInfo getManCenterPhoneHomeInfo(MCResponse mcResponse) {
+        InputStream inputStream = null;
+        InputStreamReader reader = null;
+        MCPhoneHomeInfo mcPhoneHomeInfo = null;
+        try {
+            inputStream = mcResponse.getInputStream();
+            reader = new InputStreamReader(inputStream, "UTF-8");
+            JsonObject mcPhoneHomeInfoJson = JsonValue.readFrom(reader).asObject();
+            final String version = JsonUtil.getString(mcPhoneHomeInfoJson, "mcVersion");
+            final String license = JsonUtil.getString(mcPhoneHomeInfoJson, "mcLicense", null);
+            final int responseCode = mcResponse.getResponseCode();
+            mcPhoneHomeInfo = new MCPhoneHomeInfo(version, license, responseCode);
+        } catch (Exception ignored) {
+            EmptyStatement.ignore(ignored);
+        } finally {
+            IOUtil.closeResource(reader);
+            IOUtil.closeResource(inputStream);
+        }
+        return mcPhoneHomeInfo;
+    }
+
+    private static class MCRequest implements Callable<MCResponse> {
+
+        private Node hazelcastNode;
+
+        public MCRequest(Node hazelcastNode) {
+            this.hazelcastNode = hazelcastNode;
+        }
+
+        @Override
+        public MCResponse call() throws Exception {
+            ManagementCenterConfig managementCenterConfig = hazelcastNode.config.getManagementCenterConfig();
+            String manCenterURL = managementCenterConfig.getUrl();
+            manCenterURL = manCenterURL.endsWith("/") ? manCenterURL : manCenterURL + '/';
+            URL manCenterPhoneHomeURL = new URL(manCenterURL + "phoneHome.do");
+            ManagementCenterConnectionFactory connectionFactory
+                    = hazelcastNode.getNodeExtension().getManagementCenterConnectionFactory();
```

Comment:
ConnectionFactory != null doesn't guarantee properly constructed factory, in other words, you are not creating a valid SSL connection if `https` is used; leaks sensitive information.
Also, i believe its time to implement a DefaultConnectionFactory (for DefaultNodeExtention) which would simply do the `url.openConnection()` thus we can remove this ternary operator from here & MC Service class. (minor comment, WDYT?)

Reply:
>@tkountis Did you mean I need to add this init method? --> connectionFactory.init(managementCenterConfig.getMutualAuthConfig());

Correct, without this, then even though you might have HTTPS in your config, it will still do it over HTTP.